### PR TITLE
Added typings

### PR DESCRIPTION
**Pull request by Lioness100**

### CODE REVIEW

**Review by wzhouwzhou — APPROVED** [review]
LGTM, thanks!

**Review by Vetlix — APPROVED** [review]
very good, looking back, very bad

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  public constructor(public query: string);
+
+  public search(options?: Record<string, unknown>, apiKey?: KeyData): Promise<this>;
+  public newPage(newOptions?: Options): Promise<null | this>;
+  public prevPage(newOptions?: Options): Promise<null | this>;
+}
+
+export class YTSearcher {
+  public totalResult?: number | null;
```
Comment (wzhouwzhou) [inline index.d.ts]: Actually, I'm not sure if the searcher itself has a totalResult

**Comment by tbnritzdoge** [comment]
should consider having consistent code and actual strict typings instead of unknowns everywhere. ngtm.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  public pages?: number | null;
+  public nextPageToken?: string;
+  public prevPageToken?: string;
```
Comment (wzhouwzhou) [inline index.d.ts]: Page tokens appear in the YTSearch, not the searcher

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+export const version: number;
+export const validOptions: string[];
+
+export type Options = Record<string, unknown>;
```
Comment (tbnritzdoge) [inline index.d.ts]: Why not strictly type this so the user actually knows which options are valid and what value they provide for said key

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  id: ResultID;
+}
+
+export class YTSearch {
+  public options?: Options;
+  public timestamp: number;
+
+  public constructor(public query: string);
+
+  public search(options?: Record<string, unknown>, apiKey?: KeyData): Promise<this>;
```
Comment (tbnritzdoge) [inline index.d.ts]: consistency?

**Review by tbnritzdoge — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 3 above)
Comment (Vetlix) [inline index.d.ts]: Please make the options correspond with this: https://developers.google.com/youtube/v3/docs/search/list

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+export const version: number;
+export const validOptions: string[];
+
+export type ResultType = 'youtube#video' | 'youtube#channel' | 'youtube#playlist';
+
+interface SearchOptions {
```
Comment (Vetlix) [inline index.d.ts]: Maybe export these just in case someone needs them?

**Review by diced — APPROVED** [review]

**Comment by wzhouwzhou** [comment]
@tbnritzdoge How's it looking now?

**Review by tbnritzdoge — APPROVED** [review]
Much better, thanks for the improvements. LGTM :)

**Comment by wzhouwzhou** [comment]
Thanks all

**Pull request merged**